x-pack/plugins/apm/server/lib/service_map/get_service_anomalies.ts

```diff
@@ -104,82 +94,44 @@ export async function getServiceAnomalies({
     },
   };
 
-  const response = await ml.mlSystem.mlAnomalySearch(params, mlJobIds);
+  const [anomalyResponse, jobIds] = await Promise.all([
+    ml.mlSystem.mlAnomalySearch(params, []),
+    getMLJobIds(ml.anomalyDetectors, environment),
+  ]);
 
-  return {
-    mlJobIds,
-    serviceAnomalies: transformResponseToServiceAnomalies(
-      response as ServiceAnomaliesAggResponse
-    ),
-  };
-}
+  const typedAnomalyResponse: ESSearchResponse<
+    unknown,
+    typeof params
+  > = anomalyResponse as any;
 
-interface ServiceAnomaliesAggResponse {
-  aggregations: {
-    services: {
-      buckets: Array<{
-        key: string;
-        top_score: {
-          hits: {
-            hits: Array<{
-              sort: [number];
-              _source: {
-                job_id: string;
-                by_field_value: string;
-              } & (
-                | {
-                    record_score: number | null;
-                    result_type: 'record';
-                    actual: number[];
-                  }
-                | {
-                    result_type: 'model_plot';
-                    actual?: number;
-                  }
-              );
-            }>;
+  return {
+    mlJobIds: jobIds,
+    serviceAnomalies:
+      typedAnomalyResponse.aggregations?.services.buckets
+        .filter((bucket) => jobIds.includes(bucket.key.jobId as string))
+        .map((bucket) => {
+          const metrics = bucket.metrics.top[0].metrics;
+
+          const anomalyScore =
+            (metrics.result_type === 'record' &&
+              (metrics.record_score as number)) ||
+            0;
```

Comment:
Not sure how to read this. Should it be a ternary?

Reply:
... I see now that this is not something you introduced in this PR.

Reply:
I can still improve it though :) I think I wrote it originally as well. I'll have a look.